src/basic/bitmap.c

```diff
@@ -83,10 +84,13 @@ int bitmap_set(Bitmap *b, unsigned n) {
         offset = BITMAP_NUM_TO_OFFSET(n);
 
         if (offset >= b->n_bitmaps) {
-                if (!GREEDY_REALLOC0(b->bitmaps, b->bitmaps_allocated, offset + 1))
+                new_n_bitmaps = offset + 1;
+                if (!GREEDY_REALLOC(b->bitmaps, b->bitmaps_allocated, new_n_bitmaps))
                         return -ENOMEM;
 
-                b->n_bitmaps = offset + 1;
+                /* we need to do memzero here instead of GREEDY_REALLOC0 because the array does not get cleared on bitmap_clear */
+                memzero(b->bitmaps + b->n_bitmaps, (new_n_bitmaps - b->n_bitmaps) * sizeof(b->bitmaps[0]));
+                b->n_bitmaps = new_n_bitmaps;
```

Comment:
Nice catch! However, the better fix is to really free the memory in `bitmap_clear()` and also set `b->bitmaps_allocated = 0`. Then this function does the right thing as it stands.